kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchJobSubmission.scala

```diff
@@ -96,7 +99,14 @@ class BatchJobSubmission(
 
   override private[kyuubi] def currentApplicationInfo: Option[ApplicationInfo] = {
     // only the ApplicationInfo with non-empty id is valid for the operation
-    applicationManager.getApplicationInfo(builder.clusterManager(), batchId).filter(_.id != null)
+    val applicationInfo =
+      applicationManager.getApplicationInfo(builder.clusterManager(), batchId).filter(_.id != null)
+    applicationInfo.foreach { _ =>
+      if (_appSubmissionTime <= 0) {
+        _appSubmissionTime = System.currentTimeMillis()
```

Comment:
Seems it's the `appStartTime` ? Submisstion time is the time before submiting succeed.

Reply:
why we donot set it in `submitAndMonitorBatchJob` method ?

Reply:
> why we donot set it in submitAndMonitorBatchJob method ?

Because in the `submitAndMonitorBatchJob` method, there is interval to track the app state.
And for the `getBatch` api, it will also call currentApplicationState to build the batch report, to keep the state align, it is better to update the app submission time in `currentApplicationState`.

Reply:
I think we should make state machine and the time trace consistent.

If the submitssion time means when the state changes to running, then we should update the time around this code
```
setStateIfNotCanceled(OperationState.RUNNING)
```

Otherwise the client may get the batch info with `submisstion time = 0 but state is running` or `submisstion time != 0 but state is pending`.

Reply:
for batch report, it is consistent with the appId/appState/appState now.

Reply:
We can treat it as a filed of batch application info.

Reply:
here is the pr for followup https://github.com/apache/kyuubi/pull/4129